provisioner .EnvVarFile breaks with array of scripts

#### Overview of the Issue

I have a `template.json` which in some provisioner steps, specifies an `execute_command` with `{{.EnvVarFile}}` together with setting `use_env_var_file` true; when the `scripts` item is an array of scripts, then the **second** script fails with being unable to source the varfile.

This worked when I last touched this when @SwampDragons added support after Packer 1.2.5 with PR https://github.com/hashicorp/packer/pull/6636 to solve https://github.com/hashicorp/packer/issues/6522 but I just tried using Packer again for the first time in a while, v1.5.1, and it's broken.

`PACKER_LOG=1` output confirms: the same value of `{{.EnvVarFile}}` is passed to each script but the file on disk is deleted after the first script.

#### Reproduction Steps

`template.json` containing:

```
  "provisioners": [
    {
      "execute_command": "chmod +x {{ .Path }}; sudo -E sh -c '. {{.EnvVarFile}} && {{ .Path }}'",
      "expect_disconnect": true,
      "scripts": [
        "scripts/00_pre_update",
        "scripts/01_system_update"
      ],
      "use_env_var_file": true,
      "type": "shell"
    },
[...]
```

Running packer:

```
==> amazon-ebs: Connected to SSH!
==> amazon-ebs: Provisioning with shell script: scripts/00_pre_update
    amazon-ebs: Waiting for cloud-init ... [0]
    amazon-ebs: Waiting for cloud-init ... [1]
    amazon-ebs: Waiting for cloud-init ... [2]
    amazon-ebs: Waiting for cloud-init ... [3]
    amazon-ebs: Waiting for cloud-init ... [4]
    amazon-ebs: Waiting for cloud-init ... [5]
    amazon-ebs: Mon Jan 20 21:33:56 UTC 2020
==> amazon-ebs: Provisioning with shell script: scripts/01_system_update
==> amazon-ebs: sh: 1: .: Can't open /tmp/varfile_3996.sh
==> amazon-ebs: Provisioning step had errors: Running the cleanup provisioner, if present...
```

So whatever value `{{.EnvVarFile}}` had for the first invocation was correct at the time of executing the first invocation.  By the time the second script was run, whatever value was supplied to that second invocation was wrong.

### Packer version

`Packer v1.5.1`

### Simplified Packer Buildfile

The relevant sample is above in the reproduction step.  `packer-ebs` builder.

### Operating system and Environment details

Machine invoking Packer:  Ubuntu Eoan (19.10) on amd64.
Remote machine: Ubuntu Bionic LTS (filter match for `ubuntu/images/*/ubuntu-bionic-18.04-amd64-server-*`)

### Log Fragments and crash.log files

Short log: above in reproduction steps.

Running with PACKER_LOG=1, we see that indeed the varfile is being deleted after the first script, then passed to the second script unchanged.

<details>
<summary>PACKER_LOG=1 output showing too-early deletion of variables temp file</summary>

```
2020/01/20 17:03:27 packer_1.5.1 plugin: [DEBUG] Opening new ssh session
2020/01/20 17:03:27 packer_1.5.1 plugin: [DEBUG] starting remote command: chmod +x /tmp/script_5546.sh; sudo -E sh -c '. /tmp/varfile_9722.sh && /tmp/script_5546.sh'
    amazon-ebs: Waiting for cloud-init ... [0]
    amazon-ebs: Waiting for cloud-init ... [1]
    amazon-ebs: Waiting for cloud-init ... [2]
    amazon-ebs: Waiting for cloud-init ... [3]
    amazon-ebs: Waiting for cloud-init ... [4]
2020/01/20 17:03:32 packer_1.5.1 plugin: [INFO] RPC endpoint: Communicator ended with: 0
2020/01/20 17:03:32 [INFO] 184 bytes written for 'stdout'
    amazon-ebs: Mon Jan 20 22:03:32 UTC 2020
2020/01/20 17:03:32 [INFO] 0 bytes written for 'stderr'
2020/01/20 17:03:32 [INFO] RPC client: Communicator ended with: 0
2020/01/20 17:03:32 [INFO] RPC endpoint: Communicator ended with: 0
2020/01/20 17:03:32 packer_1.5.1 plugin: [INFO] 184 bytes written for 'stdout'
2020/01/20 17:03:32 packer_1.5.1 plugin: [INFO] 0 bytes written for 'stderr'
2020/01/20 17:03:32 packer_1.5.1 plugin: [INFO] RPC client: Communicator ended with: 0
2020/01/20 17:03:32 packer_1.5.1 plugin: [DEBUG] Opening new ssh session
2020/01/20 17:03:32 packer_1.5.1 plugin: [DEBUG] starting remote command: rm -f /tmp/script_5546.sh
2020/01/20 17:03:32 packer_1.5.1 plugin: [INFO] RPC endpoint: Communicator ended with: 0
2020/01/20 17:03:32 [INFO] RPC client: Communicator ended with: 0
2020/01/20 17:03:32 [INFO] RPC endpoint: Communicator ended with: 0
2020/01/20 17:03:32 packer_1.5.1 plugin: [INFO] RPC client: Communicator ended with: 0
2020/01/20 17:03:32 packer_1.5.1 plugin: [DEBUG] Opening new ssh session
2020/01/20 17:03:32 packer_1.5.1 plugin: [DEBUG] starting remote command: rm -f /tmp/varfile_9722.sh
2020/01/20 17:03:32 packer_1.5.1 plugin: [INFO] RPC endpoint: Communicator ended with: 0
2020/01/20 17:03:32 [INFO] RPC client: Communicator ended with: 0
2020/01/20 17:03:32 [INFO] RPC endpoint: Communicator ended with: 0
2020/01/20 17:03:32 packer_1.5.1 plugin: [INFO] RPC client: Communicator ended with: 0
==> amazon-ebs: Provisioning with shell script: scripts/01_system_update
2020/01/20 17:03:32 packer_1.5.1 plugin: Opening scripts/01_system_update for reading
2020/01/20 17:03:32 packer_1.5.1 plugin: [DEBUG] Opening new ssh session
2020/01/20 17:03:32 packer_1.5.1 plugin: [INFO] 5037 bytes written for 'uploadData'
2020/01/20 17:03:32 [INFO] 5037 bytes written for 'uploadData'
2020/01/20 17:03:32 packer_1.5.1 plugin: [DEBUG] Starting remote scp process:  scp -vt /tmp
2020/01/20 17:03:32 packer_1.5.1 plugin: [DEBUG] Started SCP session, beginning transfers...
2020/01/20 17:03:32 packer_1.5.1 plugin: [DEBUG] Copying input data into temporary file so we can read the length
2020/01/20 17:03:32 packer_1.5.1 plugin: [DEBUG] scp: Uploading script_5546.sh: perms=C0644 size=5037
2020/01/20 17:03:32 packer_1.5.1 plugin: [DEBUG] SCP session complete, closing stdin pipe.
2020/01/20 17:03:32 packer_1.5.1 plugin: [DEBUG] Waiting for SSH session to complete.
2020/01/20 17:03:32 packer_1.5.1 plugin: [DEBUG] scp stderr (length 32): Sink: C0644 5037 script_5546.sh
2020/01/20 17:03:32 packer_1.5.1 plugin: [DEBUG] Opening new ssh session
2020/01/20 17:03:32 packer_1.5.1 plugin: [DEBUG] starting remote command: chmod 0755 /tmp/script_5546.sh
2020/01/20 17:03:32 packer_1.5.1 plugin: [INFO] RPC endpoint: Communicator ended with: 0
2020/01/20 17:03:32 [INFO] RPC client: Communicator ended with: 0
2020/01/20 17:03:32 [INFO] RPC endpoint: Communicator ended with: 0
2020/01/20 17:03:32 packer_1.5.1 plugin: [INFO] RPC client: Communicator ended with: 0
2020/01/20 17:03:32 packer_1.5.1 plugin: [DEBUG] Opening new ssh session
2020/01/20 17:03:32 packer_1.5.1 plugin: [DEBUG] starting remote command: chmod +x /tmp/script_5546.sh; sudo -E sh -c '. /tmp/varfile_9722.sh && /tmp/script_5546.sh'
2020/01/20 17:03:32 packer_1.5.1 plugin: [ERROR] Remote command exited with '2': chmod +x /tmp/script_5546.sh; sudo -E sh -c '. /tmp/varfile_9722.sh && /tmp/script_5546.sh'
2020/01/20 17:03:32 packer_1.5.1 plugin: [INFO] RPC endpoint: Communicator ended with: 2
2020/01/20 17:03:32 [INFO] 0 bytes written for 'stdout'
2020/01/20 17:03:32 [INFO] 42 bytes written for 'stderr'
==> amazon-ebs: sh: 1: .: Can't open /tmp/varfile_9722.sh
2020/01/20 17:03:32 [INFO] RPC client: Communicator ended with: 2
2020/01/20 17:03:32 [INFO] RPC endpoint: Communicator ended with: 2
```
</details>

--- a/provisioner/shell/provisioner.go
+++ b/provisioner/shell/provisioner.go
@@ -129,8 +129,7 @@ func (p *Provisioner) Prepare(raws ...interface{}) error {
 	}
 
 	if p.config.RemotePath == "" {
-		p.config.RemotePath = fmt.Sprintf(
-			"%s/%s", p.config.RemoteFolder, p.config.RemoteFile)
+		p.config.RemotePath = fmt.Sprintf("%s/%s", p.config.RemoteFolder, p.config.RemoteFile)
 	}
 
 	if p.config.Scripts == nil {
@@ -232,7 +231,6 @@ func (p *Provisioner) Provision(ctx context.Context, ui packer.Ui, comm packer.C
 		}
 
 		p.config.envVarFile = tf.Name()
-		defer os.Remove(p.config.envVarFile)
 
 		// upload the var file
 		var cmd *packer.RemoteCmd
@@ -256,9 +254,7 @@ func (p *Provisioner) Provision(ctx context.Context, ui packer.Ui, comm packer.C
 				Command: fmt.Sprintf("chmod 0600 %s", remoteVFName),
 			}
 			if err := comm.Start(ctx, cmd); err != nil {
-				return fmt.Errorf(
-					"Error chmodding script file to 0600 in remote "+
-						"machine: %s", err)
+				return fmt.Errorf("Error chmodding script file to 0600 in remote machine: %s", err)
 			}
 			cmd.Wait()
 			p.config.envVarFile = remoteVFName
@@ -345,19 +341,22 @@ func (p *Provisioner) Provision(ctx context.Context, ui packer.Ui, comm packer.C
 			return err
 		}
 
-		if !p.config.SkipClean {
+		if p.config.SkipClean {
+			continue
+		}
 
-			// Delete the temporary file we created. We retry this a few times
-			// since if the above rebooted we have to wait until the reboot
-			// completes.
-			err = p.cleanupRemoteFile(p.config.RemotePath, comm)
-			if err != nil {
-				return err
-			}
-			err = p.cleanupRemoteFile(p.config.envVarFile, comm)
-			if err != nil {
-				return err
-			}
+		// Delete the temporary file we created. We retry this a few times
+		// since if the above rebooted we have to wait until the reboot
+		// completes.
+		if err := p.cleanupRemoteFile(p.config.RemotePath, comm); err != nil {
+			return err
+		}
+
+	}
+
+	if !p.config.SkipClean {
+		if err := p.cleanupRemoteFile(p.config.envVarFile, comm); err != nil {
+			return err
 		}
 	}
 

